src/components/Asset/Edit/AddService.tsx

```diff
@@ -0,0 +1,270 @@
+import { mapTimeoutStringToSeconds, normalizeFile } from '@utils/ddo'
+import content from '../../../../content/pages/editService.json'
+import { useAbortController } from '@hooks/useAbortController'
+import EditFeedback from './EditFeedback'
+import { useAsset } from '@context/Asset'
+import { setNftMetadata } from '@utils/nft'
+import { getEncryptedFiles } from '@utils/provider'
+import { useAccount, useNetwork, useSigner } from 'wagmi'
+import { transformConsumerParameters } from '@components/Publish/_utils'
+import { marketFeeAddress, publisherMarketFixedSwapFee } from 'app.config'
+import { ethers } from 'ethers'
+import FormAddService from './FormAddService'
+import { transformComputeFormToServiceComputeOptions } from '@utils/compute'
+import { useCancelToken } from '@hooks/useCancelToken'
+import { serviceValidationSchema } from './_validation'
+import DebugEditService from './DebugEditService'
+import styles from './index.module.css'
+import { useUserPreferences } from '@context/UserPreferences'
+
+export default function AddService({
+  asset
+}: {
+  asset: AssetExtended
+}): ReactElement {
+  const { debug } = useUserPreferences()
+  const { fetchAsset, isAssetNetwork } = useAsset()
+  const { address: accountId } = useAccount()
+  const { chain } = useNetwork()
+  const { data: signer } = useSigner()
+  const newAbortController = useAbortController()
+  const newCancelToken = useCancelToken()
+
+  const [success, setSuccess] = useState<string>()
+  const [error, setError] = useState<string>()
+  const hasFeedback = error || success
+
+  // add new service
+  async function handleSubmit(values: ServiceEditForm, resetForm: () => void) {
+    try {
+      if (!isAssetNetwork) {
+        setError('Please switch to the correct network.')
+        return
+      }
+
+      // --------------------------------------------------
+      // 1. Create Datatoken
+      // --------------------------------------------------
+      const nft = new Nft(signer)
+
+      const datatokenAddress = await nft.createDatatoken(
+        asset.nftAddress,
+        accountId,
+        accountId,
+        values.paymentCollector,
+        marketFeeAddress,
+        process.env.NEXT_PUBLIC_OCEAN_TOKEN_ADDRESS,
+        publisherMarketFixedSwapFee,
+        '100000000',
+        'DataToken',
+        'DT',
+        1
+      )
+
+      LoggerInstance.log('Datatoken created.', datatokenAddress)
+
+      // --------------------------------------------------
+      // 2. Create Pricing
+      // --------------------------------------------------
+      const datatoken = new Datatoken(signer)
+
+      let pricingTransactionReceipt
+      if (values.price > 0) {
+        LoggerInstance.log(
+          `Creating fixed rate exchange with price ${values.price} for datatoken ${datatokenAddress}`
+        )
+
+        const freParams: FreCreationParams = {
+          fixedRateAddress: process.env.NEXT_PUBLIC_FIXED_RATE_EXCHANGE_ADDRESS,
```

Comment:
should this get from config instead of process.env?

Reply:
good idea, thanks 🙂  fixed